storage/gcs/src/main/java/io/aiven/kafka/tieredstorage/storage/gcs/CredentialsBuilder.java

```diff
@@ -32,12 +32,28 @@ private CredentialsBuilder() {
         // hide constructor
     }
 
+    public static void validate(final Boolean defaultCredentials,
+                                final String credentialsJson,
+                                final String credentialsPath) {
+        final int nonNulls =
+            (defaultCredentials != null ? 1 : 0)
+            + (credentialsJson != null ? 1 : 0)
+            + (credentialsPath != null ? 1 : 0);
```

Comment:
Does performance matter here? If no, I'd say `Stream.of(...).filter(Objects::nonNull).count()` is a little bit more readable

Reply:
Yeah, that's better